### DISCUSSION TITLE
How to add endpoint dynamicly after startup

### DISCUSSION THREAD

**Discussion by iplusMario** [dotnet/aspnetcore#26170] [General]
Is it possible to add new enpoint dynamicly after startup executed in AspNetCore?

Any suggestions how would you solve this particular problem which is simplified.

We are having a admin controller which can manage how many chat rooms are open.
Each chat room has its own WebSocket or Hub if singalr which is registered to his own endpoint ("/hub1", "/hub2", etc.).
So admin connects to controller calls action OpenChatRoom("/Hub3") in which it would need to registers new
endpoint "hub3" which uses websocket handler/middelware in runtime and does not need to restart asp net core app.

**Comment by Tratcher** [comment]
It's not easy. You need to implement an EndpointDataSource like this example.
https://github.com/microsoft/reverse-proxy/blob/be99c24e78c34f5eb73c69212c20ce16a00e0367/src/ReverseProxy/Service/Management/ProxyConfigManager.cs#L31
It sounds like your using SignalR? I don't know if this would work with SignalR.
